lib/foreman/silenced_logger.rb

```diff
@@ -1,15 +1,38 @@
 # Wrap a ::Logging::Logger and implement #silence to temporarily increase the
 # min log level to execute a block of code. Used by sprockets-rails' quiet
-# assets logging feature.
+# assets logging feature. Foreman uses logging gem instead Rails logging stack
+# and it does not provide silencing feature (ships with just a noop method).
 module Foreman
   class SilencedLogger < SimpleDelegator
-    def silence(new_level = Logger::ERROR, &block)
-      old_level = self.level
+    def initialize(*args)
```

Comment:
I can do that with one big warning, `SilencedLogger` is not a singleton, we create instance for every logger (app, sql, etc). To prevent threading issues, I would need to create the symbol key as something like `foreman_log_level_#{object_id}` and that leads to memory leak on older Ruby VMs, according to discussion at https://github.com/rails/rails/pull/20507 but this should be solved with Ruby 2.2. It is acceptable for us as we are moving to 2.2+ in production now I guess: https://www.sitepoint.com/symbol-gc-ruby-2-2/ and also it is a technique that other part of Rails use: https://github.com/rails/activerecord-session_store/blob/master/lib/active_record/session_store/extension/logger_silencer.rb

The implementation I propose and which is copy-pasted from Rails 5 is safe on all Ruby versions.

I saw @swapab in the PR mentioned, opinion?

Reply:
@lzap I think @iNecas 's code is enough in our scenario - we only test with Ruby 2.3 & 2.4 in Jenkins even

Reply:
@dLobatog not sure what you mean, I cannot do just `Thread.current[:foreman_silenced_logger_level]` as we create multiple instances of this class *per-thread*. I lean towards the most safe hash approach.

Reply:
Btw. you were using `@@class_variavles` in the original code, so the code would not handle the mutliple instances of the class in single thread anyway, if you didn't use something like the `object_id` in the key.